privacyidea/lib/eventhandler/scripthandler.py

```diff
@@ -164,14 +180,26 @@ def do(self, action, options=None):
             proc_args.append("--logged_in_role")
             proc_args.append(logged_in_user.get("role", "none"))
 
-        try:
-            p = subprocess.Popen(proc_args, cwd=self.script_directory)
-            log.info("Started script {script!r}:"
-                     " {process!r}".format(script=script_name, process=p))
-        except Exception as e:
-            log.warning("Failed to execute script {0!r}: {1!r}".format(
-                script_name, e))
-            log.warning(traceback.format_exc())
+        if handler_options.get("background", SCRIPT_BACKGROUND) == SCRIPT_BACKGROUND:
+            try:
+                p = subprocess.Popen(proc_args, cwd=self.script_directory)
+                log.info("Started script {script!r} in background:"
+                         " {process!r}".format(script=script_name, process=p))
+            except Exception as e:
+                log.warning("Failed to execute script {0!r}: {1!r}".format(
+                    script_name, e))
+                log.warning(traceback.format_exc())
+
+        elif handler_options.get("background") == SCRIPT_WAIT:
+            try:
+                log.info("Starting script {script!r} in foreground.".format(script=script_name))
+                r = subprocess.check_call(proc_args, cwd=self.script_directory)
+            except Exception as e:
+                log.warning("Failed to execute script {0!r}: {1!r}".format(
+                    script_name, e))
+                log.warning(traceback.format_exc())
+                if handler_options.get("raise_error"):
+                    raise e
 
```

Comment:
Should we add an ``else`` branch like
```python
else:
    log.warning("Invalid value: {!r}".format(handler_options.get("background")))
```
?

Reply:
No, but I will add an else branch, if raise_error is not set.
Then it would be great to have this merged ;-)